src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private T[] elements;
+    private int size;
+    private static final int INITIAL_CAPACITY = 10;
+
+    @SuppressWarnings("unchecked")
+    public MateStack() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        ensureCapacity();
+        elements[size++] = value;
     }
```

Comment:
It's a good practice to separate the logic of adding a value and ensuring the capacity of the stack. You can create a separate private method `addValue(T value)` that will add a value to the stack. Then, in the `push(T value)` method, you can first call `ensureCapacity()` and then `addValue(T value)`. This will make your code more modular and easier to read and maintain.